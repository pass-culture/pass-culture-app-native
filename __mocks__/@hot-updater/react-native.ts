import { ReactNode } from 'react'

const getUpdateSource = jest.fn(() => Promise.resolve({ source: 'source' }))

const HotUpdater = jest.fn(() => (app: ReactNode) => app)

export default Object.assign(HotUpdater, {
  getUpdateSource,
})
