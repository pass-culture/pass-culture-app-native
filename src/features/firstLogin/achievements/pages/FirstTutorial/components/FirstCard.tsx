import { t } from '@lingui/macro'
import React from 'react'

import { _ } from 'libs/i18n'
import TutorialPassLogo from 'ui/animations/tutorial_pass_logo.json'
import {
  AchievementCardKeyProps,
  GenericAchievementCard,
} from 'ui/components/achievements/components/GenericAchievementCard'

export function FirstCard(props: AchievementCardKeyProps) {
  function onButtonPress() {
    props.swiperRef?.current?.goToNext()
  }

  return (
    <GenericAchievementCard
      animation={TutorialPassLogo}
      buttonCallback={onButtonPress}
      buttonText={_(t`Continuer`)}
      pauseAnimationOnRenderAtFrame={62}
      subTitle={_(t`c'est...`)}
      text={_(t`une initiative financée par le Ministère de la Culture.`)}
      title={_(t`Le pass Culture`)}
      swiperRef={props.swiperRef}
      name={props.name}
      index={props.index}
      activeIndex={props.activeIndex}
    />
  )
}
