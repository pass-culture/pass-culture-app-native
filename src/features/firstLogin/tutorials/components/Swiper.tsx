import React from 'react'
import GestureRecognizer from 'react-native-swipe-gestures'

const CONFIG = {
  velocityThreshold: 0.03,
  directionalOffsetThreshold: 400,
  gestureIsClickThreshold: 0.1,
}

type Props = Omit<GestureRecognizer['props'], 'config'>

export const Swiper: React.FunctionComponent<Props> = (props) => {
  return (
    <GestureRecognizer {...props} config={CONFIG}>
      {props.children}
    </GestureRecognizer>
  )
}
