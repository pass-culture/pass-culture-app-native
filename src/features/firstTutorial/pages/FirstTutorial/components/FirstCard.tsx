import { t } from '@lingui/macro'
import React from 'react'

import TutorialPassLogo from 'ui/animations/tutorial_pass_logo.json'
import {
  AchievementCardKeyProps,
  GenericAchievementCard,
} from 'ui/components/achievements/components/GenericAchievementCard'
import { Helmet } from 'ui/web/global/Helmet'

export function FirstCard(props: AchievementCardKeyProps) {
  function onButtonPress() {
    props.swiperRef?.current?.goToNext()
  }

  const helmetTitle =
    t`Étape ${(props.activeIndex || 0) + 1} sur ${
      (props.lastIndex || 3) + 1
    } | Tutoriel "Comment ça marche"` + ' | pass Culture'

  return (
    <React.Fragment>
      <Helmet title={helmetTitle} />
      <GenericAchievementCard
        animation={TutorialPassLogo}
        buttonCallback={onButtonPress}
        buttonText={t`Continuer`}
        pauseAnimationOnRenderAtFrame={62}
        subTitle={t`c'est...`}
        text={t`une initiative du Gouvernement financée par le ministère de la Culture.`}
        title={t`Le pass Culture`}
        swiperRef={props.swiperRef}
        name={props.name}
        index={props.index}
        activeIndex={props.activeIndex}
        lastIndex={props.lastIndex}
      />
    </React.Fragment>
  )
}
