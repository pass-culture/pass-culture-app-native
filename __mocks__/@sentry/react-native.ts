import * as ActualSentryModule from '@sentry/react-native'

const MockedSentryModule = jest.genMockFromModule<typeof ActualSentryModule>('@sentry/react-native')

module.exports = {
  ...MockedSentryModule,
}
