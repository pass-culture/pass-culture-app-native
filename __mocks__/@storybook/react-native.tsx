import React from 'react'; // @storybook
import { View } from 'react-native';

const configure = jest.fn();
const getStorybookUI = jest.fn().mockImplementation(() => () => <View />);

export { configure, getStorybookUI };
