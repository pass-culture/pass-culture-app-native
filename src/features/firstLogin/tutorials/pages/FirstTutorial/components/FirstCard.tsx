import { t } from '@lingui/macro'
import React from 'react'

import { CardKey, GenericCard } from 'features/firstLogin/tutorials/components/GenericCard'
import { _ } from 'libs/i18n'
import { storage } from 'libs/storage'
import TutorialPassLogo from 'ui/animations/tutorial_pass_logo.json'

export function FirstCard(props: CardKey) {
  function onButtonPress() {
    storage.saveObject('has_seen_tutorials', true)
    props.swiperRef?.current?.goToNext()
  }
  return (
    <GenericCard
      animation={TutorialPassLogo}
      buttonCallback={onButtonPress}
      buttonText={_(t`Continuer`)}
      pauseAnimationOnRenderAtFrame={62}
      subTitle={_(t`c'est...`)}
      text={_(t`une initiative financée par le Ministère de la Culture.`)}
      title={_(t`Le pass Culture`)}
      {...props}
    />
  )
}
