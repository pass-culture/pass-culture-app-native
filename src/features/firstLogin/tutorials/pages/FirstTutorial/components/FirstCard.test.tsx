import AsyncStorage from '@react-native-community/async-storage'
import { act, fireEvent, render } from '@testing-library/react-native'
import React, { RefObject } from 'react'
import Swiper from 'react-native-web-swiper'

import { FirstCard } from './FirstCard'

describe('FirstCard', () => {
  it('should render first card', () => {
    const firstTutorial = render(<FirstCard />)
    expect(firstTutorial).toMatchSnapshot()
  })
  it('should swipe to next card on button press', async () => {
    const ref = { current: { goToNext: jest.fn() } }
    const { getByText } = render(<FirstCard swiperRef={(ref as unknown) as RefObject<Swiper>} />)
    const button = await getByText('Continuer')
    fireEvent.press(button)
    expect(AsyncStorage.setItem).toBeCalledWith('has_seen_tutorials', 'true')
    await act(async () => {
      expect(ref.current.goToNext).toHaveBeenCalled()
    })
  })
})
