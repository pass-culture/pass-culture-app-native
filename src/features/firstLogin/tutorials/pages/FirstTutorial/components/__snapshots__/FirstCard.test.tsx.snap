// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstCard should render first card 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingHorizontal": 20,
      },
    ]
  }
>
  <View
    flex={2}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 2,
          "flexShrink": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexGrow": 1,
        },
      ]
    }
  >
    <View>
      undefined-Lottie-Mock
    </View>
  </View>
  <View
    flex={1}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#151515",
          "fontFamily": "Montserrat-ExtraBoldItalic",
          "fontSize": 28,
          "lineHeight": 34,
          "textAlign": "center",
        },
      ]
    }
  >
    Le pass Culture
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#151515",
          "fontFamily": "Montserrat-MediumItalic",
          "fontSize": 24,
          "lineHeight": 28,
          "textAlign": "center",
        },
      ]
    }
  >
    c'est...
  </Text>
  <View
    flex={1}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#151515",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 15,
          "lineHeight": 20,
          "textAlign": "center",
        },
      ]
    }
  >
    une initiative financée par le Ministère de la Culture.
  </Text>
  <View
    flex={2}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 2,
          "flexShrink": 1,
        },
      ]
    }
  />
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#eb0055",
          "borderRadius": 24,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "maxWidth": 500,
          "opacity": 1,
          "paddingBottom": 2,
          "paddingLeft": 2,
          "paddingRight": 2,
          "paddingTop": 2,
          "width": "100%",
        }
      }
      testID="button-container"
    >
      <Text
        adjustsFontSizeToFit={false}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Montserrat-Bold",
              "fontSize": 15,
              "lineHeight": 20,
              "marginLeft": 5,
              "maxWidth": 650,
            },
          ]
        }
        testID="button-title"
        textColor="#ffffff"
      >
        Continuer
      </Text>
    </View>
  </View>
  <View
    flex={3}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 3,
          "flexShrink": 1,
        },
      ]
    }
  />
</View>
`;
